x/stakeibc/types/gov.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"fmt"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	ProposalTypeAddValidator = "AddValidator"
+)
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeAddValidator)
+	govtypes.RegisterProposalTypeCodec(&AddValidatorProposal{}, "stakeibc/AddValidatorProposal")
+}
+
+var (
+	_ govtypes.Content = &AddValidatorProposal{}
+)
+
+func NewAddValidatorProposal(title, description, hostZone, name, address string) govtypes.Content {
+	return &AddValidatorProposal{
+		Title:            title,
+		Description:      description,
+		HostZone:         hostZone,
+		ValidatorName:    name,
+		ValidatorAddress: address,
+	}
+}
+
+func (p *AddValidatorProposal) GetTitle() string { return p.Title }
+
+func (p *AddValidatorProposal) GetDescription() string { return p.Description }
+
+func (p *AddValidatorProposal) ProposalRoute() string { return RouterKey }
+
+func (p *AddValidatorProposal) ProposalType() string {
+	return ProposalTypeAddValidator
+}
+
+func (p *AddValidatorProposal) ValidateBasic() error {
+	err := govtypes.ValidateAbstract(p)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Add more checks?
```

Comment:
not sure what else we would add? maybe verify that "valoper" shows up in validator address?